### PR TITLE
Build VICE 3.5 GTK3, not the legacy SDL2 version.

### PR DESCRIPTION
**Pull request by dqh-au**
Also update compile time features to match official VICE macOS builds. (VICE dev team member here)

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by carlocab** [comment]
Thanks for opening this. Is there any way the features you've enabled via `configure` flags could be set by default, perhaps after a check for availability?

Typically formulae are updated by someone who knows very little about the actual package being installed. This makes maintaining formulae with a lot of custom compiler flags quite difficult to do. (I'm sure there are plenty of formulae in homebrew-core with outdated compiler flags that currently do nothing, and present a risk of breakage at some point.)

**Comment by dqh-github** [comment]
In theory it is of course possible, but the build system for it has evolved over nearly 30 years and changes to it need to be made slowly and carefully. Additionally, VICE is not just a macOS project, but also Linux, BSD, Windows (via msys2), and other voices would prefer that VICE build only the specifically requested features.

In this particular case, even if we had your proposed change in place, the brew formula would still have needed several changes from someone familiar with our development priorities - because we are migrating to GTK3. A version bump pullreq would still have happily continued building the old SDL2 UI based on GTK3 being absent and SDL2 being present.

The UI migration has been an unusually large change for the project. From here on in I expect that user submitted version bump pullreqs will be just fine, just as they been previously. But I think it should be building the same set of features that our official macOS and Windows binary distributions have rather than a minimal subset.

**Comment by carlocab** [comment]
Sorry to push you on this, but if they're already the default features on two platforms, doesn't it make more sense to enable them by default, but allow for `--disable-x` and `--without-y` flags for those who want to customise their builds?

Admittedly, I know the above isn't a change that's going to be made overnight, so maybe not quite relevant for the current PR. Still interested in your thoughts though.

While you're here: is there any way you can update the test so it does something more substantial? We don't really like tests that just do `--help` or `--version`. Cf. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

File: Formula/vice.rb [hunk 1]
```diff
@@ -19,56 +20,50 @@ class Vice < Formula
...
-  def caveats
-    <<~EOS
-      App bundles are no longer built for each emulator. The binaries are
-      available in #{HOMEBREW_PREFIX}/bin directly instead.
-    EOS
-  end
-
   test do
-    assert_match "Usage", shell_output("#{bin}/petcat -help", 1)
+    assert_match /.*cycle limit reached.*/, shell_output("#{bin}/x64sc -console -limitcycles 1000000 -logfile -", 1)
```
Comment (carlocab) [inline Formula/vice.rb:67]: Thanks for updating the test. I think you don't need a regex for this:
```suggestion
    assert_match "cycle limit reached", shell_output("#{bin}/x64sc -console -limitcycles 1000000 -logfile -", 1)
```

**Comment by dqh-github** [comment]
I've updated the test to emulate a million cycles (~1 second) of a Commodore 64 and then exit with an expected status code and include specific output in the stdout log. Hopefully that works for you.

The enabled features are those that we think should be set for users that can't or don't build VICE themselves. Linux and *BSD are the primary development targets of this project by the number and seniority of VICE developers that use it, and they don't want VICE to magically configure itself based on the available libraries.

One thing that might be viable is a new --standard-bindist flag that sets other flags as we'd like them for binary distributions, this would simplify several internal scenarios internally as well. We'll discuss adding this, however since we typically release a new versioned tarball once per year just before Christmas, nothing is going to happen in time for this pull request to be useful to anyone.

You asked for my thoughts. To be honest with you, specifically regarding the configure flags, I am sure your intentions are good but it seems inappropriate to me that you're trying to use your position to force other open source projects to organise themselves in a way that suits you.

**Comment by carlocab** [comment]
Thanks again! One last request (from me, at least): could you change your commit message to something like

    vice: use gtk+3, match official build

Anything that puts the formula name first should be fine.

**Comment by carlocab** [comment]
I missed your last comment:

>One thing that might be viable is a new --standard-bindist flag that sets other flags as we'd like them for binary distributions, this would simplify several internal scenarios internally as well. We'll discuss adding this, however since we typically release a new versioned tarball once per year just before Christmas, nothing is going to happen in time for this pull request to be useful to anyone.

A `--standard-bindist` flag would be ideal. Of course, as I suggested, I'm not asking for this to be implemented right away.

>You asked for my thoughts. To be honest with you, specifically regarding the configure flags, I am sure your intentions are good but it seems inappropriate to me that you're trying to use your position to force other open source projects to organise themselves in a way that suits you.

I'm sorry if I came across that way, but I don't think I'm forcing anyone to organise in a way that suits me. I am merely making suggestions that I think would aid in the maintenance of this formula.  Formulae that are easier to maintain are less prone to breaking, and broken formulae are often removed when no one around knows how to fix them. My suggestions are intended to benefit your users, not me.

Thanks again for this PR, and for fixing the test!

**Review by carlocab — APPROVED** [review]

**Comment by dqh-github** [comment]
Thanks for the merge.

> I'm sorry if I came across that way, but I don't think I'm forcing anyone to organise in a way that suits me. I am merely making suggestions that I think would aid in the maintenance of this formula. Formulae that are easier to maintain are less prone to breaking, and broken formulae are often removed when no one around knows how to fix them. My suggestions are intended to benefit your users, not me.

It's not a mere suggestion if it's made in the context of you having power over the distribution channel, without care being taken to point out that not complying won't prevent this or a future merge. However you intended it, with the power imbalance there is an undertone of it being up to you how our build system should work if we want to have influence how our software is configured for Homebrew users.

**Comment by carlocab** [comment]
I don't know what kind of baggage you're carrying, but don't bring that here.

You don't know me, nor are you really familiar with how this project works. Perhaps I should have been clearer about what the impact of my questions was on the status of your PR, but you should not be jumping to conclusions when your evidence for those conclusions is scant.

Just to emphasise a point I've already made above, since you seem so keen on absolute clarity, let me quote this:

> so maybe not quite relevant for the current PR

So, actually, I'm pretty sure this

> without care being taken to point out that not complying won't prevent this or a future merge

is totally unwarranted.

This PR has left a bad taste in my mouth, and I don't think you're happy about it either. However, I hope this doesn't deter you from making future PRs. I'll let the other maintainers handle them.

**Comment by dqh-github** [comment]
I'm sorry to have unsettled you but I understand your reaction. No one likes to hear that someone thinks they haven't behaved appropriately, especially when they were completely unaware of how their actions are being perceived. You can of course dismiss me as uniquely sensitive, but you could also choose to consider the possibility that my experience is not unique.

> Sorry to push you on this

> I know the above isn't a change that's going to be made overnight

> so **maybe** not quite relevant for **the current PR**

This language makes it clear that, for now, you're **allowing** us to keep our existing build system that we don't want to change. It absolutely suggests the likelihood that for a future pull request, Homebrew might decide to make this issue 'relevant' and try to force our hand. And this is what I'm talking about, I don't think Homebrew should see themselves as in a position to be **requiring** internal changes to other open source projects. Hopefully you agree with that principle even if you don't accept my interpretation of your language.

**Comment by carlocab** [comment]
>No one likes to hear that someone thinks they haven't behaved appropriately, especially when they were completely unaware of how their actions are being perceived.

You're right. This is why I was avoiding saying so, but I may as well now: you are behaving inappropriately. Perhaps you knew this and didn't care. I don't know. Unlike you, I don't want to make presumptions about what you're aware of and what your intentions are.

>This language makes it clear that

No, it really doesn't. I asked you to stop carrying your baggage into this conversation, and yet here it is again.

>I don't think Homebrew should see themselves as in a position to be requiring internal changes to other open source projects.

You're asking to make changes in this project, but are offended when someone from this project asks you if you could also make changes in yours? Geez, would you take a good, long, hard look at yourself, please?

**Comment by dqh-github** [comment]
> No, it really doesn't.

It seems that you are denying that you would ever try to force us into changing our build system by holding back a pullreq like this, in which case this is very welcome news and I apologize for misunderstanding you.

**Comment by Rhialto** [comment]
As an aside: With my package manager hat on (I maintain some packages for for pkgsrc, including VICE), I really really want configure scripts to behave predictably and repeatably, and not use libraries just because it can find them on the build system, or build without them if it can't.

**Pull request closed without merging**